### ISSUE TITLE
finish extraction of BundleContext configuration class from bundle.rb (blocked by #122)

### ISSUE THREAD

**Issue by jmartin-sul** [sul-dlss/pre-assembly#124]
the functionality lives in its own class now (`BundleContext`), but that class should also live in a file of its own.  same goes for the tests in bundle_spec.rb that were more about context and config validation than they were about the `Bundle#run_pre_assembly` functionality.

assuming #122 gets merged in close to its current form, there are a number of TODOs in bundle.rb and bundle_spec.rb that point to what should be split out to a separate file.

**Comment by ndushay** [comment]
Note that based on modeling meeting on Thurs 8/30, BundleContext should be an ActiveModel.

![IMG_2710.JPG](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/5522c457e2d8b81e00d2eea6/125516c66e82c728ace21e0d46c1b9bd1cd0d4f8dfabf301b44fff9e2c4533317346f62927d7b370e7402f5457561ba441585e10aafedf79f7b02e6c981e5ab38532022d8c6853beb2901eb60bbb25884a83b0b8834ac4ea3b22f5c6.JPG)

**Comment by ndushay** [comment]
FYI, here's what i get from loading a test_data yaml file:

```
irb(main):003:0> yf = 'spec/test_data/project_config_files/proj_revs.yaml'
=> "spec/test_data/project_config_files/proj_revs.yaml"
irb(main):004:0> pp = YAML.load_file(yf)
=> {"project_style"=>{"content_structure"=>"simple_image", "get_druid_from"=>"suri"}, "bundle_dir"=>"spec/test_data/bundle_input_a/", "staging_dir"=>"tmp", "accession_items"=>nil, "manifest"=>"manifest.csv", "checksums_file"=>"checksums.txt", "desc_md_template"=>"mods_template.xml", "progress_log_file"=>"tmp/progress_revs.yaml", "project_name"=>"Revs", "validate_files"=>true, "publish_attr"=>{"publish"=>"no", "shelve"=>"no", "preserve"=>"yes"}, "content_md_creation"=>{"style"=>"default"}, "object_discovery"=>{"use_manifest"=>true, "glob"=>nil, "regex"=>nil}, "stageable_discovery"=>{"use_container"=>true, "glob"=>nil, "regex"=>nil}, "manifest_cols"=>{"object_container"=>"filename", "source_id"=>"sourceid", "label"=>"label"}, "content_exclusion"=>nil}
```

**Comment by jmartin-sul** [comment]
i would go for `has_one`.  i don't think we want to de-dupe and share `BundleContext` instances.  or, maybe we want to run multiple jobs off the same one, for repeatable report runs (out of scope for this work cycle, but a nice enhancement for the future).  but for data model simplicity, i think it's fair to assume that a user creates a `BundleContext` instance, and no other user gets to share it.

open to arguments to the contrary, of course.  that's just my initial read on it.
